src/start.ts

```diff
@@ -54,265 +54,331 @@ export const handler = async (_event: any = {}): Promise<any> => {
-            ...node.evm.encoding.safeDecode(template.parameters),
-          };
-          const reservedParameters =
-            node.adapters.http.parameters.getReservedParameters(
-              configEndpoint,
-              apiCallParameters || {}
-            );
-          if (!reservedParameters._type) {
-            console.log(
-              "[ERROR]\treserved parameter 'type' is missing for endpoint:",
-              endpointName
-            );
-            return;
-          }
-          const sanitizedParameters: adapter.Parameters = node.utils.removeKeys(
-            apiCallParameters || {},
-            ois.RESERVED_PARAMETERS
-          );
-          const adapterApiCredentials = apiCredentials
-            .filter((c) => c.oisTitle === oisTitle)
-            .map((c) => node.utils.removeKey(c, "oisTitle"));
-
-          const options: adapter.BuildRequestOptions = {
-            ois: configOis,
-            endpointName,
-            parameters: sanitizedParameters,
-            apiCredentials: adapterApiCredentials as adapter.ApiCredentials[],
-            metadata: null,
-          };
-
-          const apiResponse = await adapter.buildAndExecuteRequest(options);
-          if (!apiResponse || !apiResponse.data) {
-            console.log(
-              "[ERROR]\tfailed to fetch data from API for endpoint:",
-              endpointName
-            );
-            return;
-          }
-          console.log("[INFO]\tAPI server response data:", apiResponse.data);
+        const rrpBeaconServerKeeperJobsByKeeperSponsor = groupBy(
+          triggers.rrpBeaconServerKeeperJobs,
+          "keeperSponsor"
+        );
+        const keeperSponsorAddresses = Object.keys(
+          rrpBeaconServerKeeperJobsByKeeperSponsor
+        );
 
-          let apiValue: ethers.BigNumber;
-          try {
-            const response = adapter.extractAndEncodeResponse(
-              apiResponse.data,
-              reservedParameters as adapter.ReservedParameters
+        const keeperSponsorWalletPromises = keeperSponsorAddresses.map(
+          async (keeperSponsor) => {
+            const keeperSponsorWallet = deriveKeeperSponsorWallet(
+              airnodeHDNode,
+              keeperSponsor,
+              provider
             );
-            apiValue = ethers.BigNumber.from(response.values[0].toString());
 
-            console.log("[INFO]\tAPI server value:", apiValue.toString());
-          } catch (error) {
-            console.log(
-              "[ERROR]\tfailed to extract value from API response:",
-              JSON.stringify(apiResponse.data)
-            );
-            let message;
-            if (error instanceof Error) message = error.message;
-            else message = String(error);
-            console.log("[DEBUG]\tmessage:", message);
-            return;
-          }
+            // Fetch keeperSponsorWallet transaction count
+            let keeperSponsorWalletTransactionCount =
```

Comment:
I probably don't have all the context and all of this is from a place of trying to help:

`let keeperSponsorWalletTransactionCount` is updated once further down (`const nonce = keeperSponsorWalletTransactionCount++;`); if you converted the for loop to a `.map` and exposed the index argument you'd be able to say `const nonce = keeperSponsorWalletTransactionCount+sponsorIndex`. If the function handed to `.map` was `async` you'd get coroutine execution, which could be very useful. This should be fine because the wallets are separate. Airnode does this and it's probably why transactions were grouped like this.

So instead of 
```
            for (const {
              templateId,
              parameters,
              oisTitle,
              endpointName,
              deviationPercentage,
              requestSponsor,
              eventLogMaxBlocks,
            } of rrpBeaconServerKeeperJobs) {
```
it could be 
```
            rrpBeaconServerKeeperJobs.map(async ({
              templateId,
              parameters,
              oisTitle,
              endpointName,
              deviationPercentage,
              requestSponsor,
              eventLogMaxBlocks}, index) => {...});
```

Reply:
I would really like to revert `keeperSponsorWalletTransactionCount` to a being a `const` instead of a `let` but each rrpBeaconServerKeeperJob must be run sequentially since these will translate into 1 or more transaction per keeperSponsorWallet (similar to [here](https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/evm/fulfillments/index.ts#L120) in node package).

I might be misunderstanding the suggested improvement tho 😄

Reply:
Haha, that makes sense, thanks for bearing with me :)